hoomd/hpmc/external/linear.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) 2009-2024 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+"""Linear external potential.
+
+.. invisible-code-block: python
+
+    simulation = hoomd.util.make_example_simulation()
+    sphere = hoomd.hpmc.integrate.Sphere()
+    sphere.shape['A'] = dict(diameter=0.0)
+    simulation.operations.integrator = sphere
+"""
+
+import hoomd
+
+from .external import External
+
+
+@hoomd.logging.modify_namespace(('hpmc', 'external', 'Linear'))
+class Linear(External):
+    """Linear external potential (HPMC).
+
+    Args:
+        default_alpha (float): Default value for alpha
+            :math:`[\\mathrm{energy}] [\\mathrm{length}]^{-1}`
+        plane_origin ([`float`, `float`, `float`]): A point on the plane of 0
+            energy.
+        plane_normal ([`float`, `float`, `float`]): A vector perpendicular to
+            the plane of 0 energy.
+
+    `Linear` computes a linear external potential on all particles in the
+    simulation state:
+
+    .. math::
+
+        U_{\\mathrm{external},i} = \\alpha_i \\cdot \\vec{n} \\cdot
+            ( \\vec{r}_i - \\vec{p} )
+
+    where :math:`\\alpha_i` (`alpha`) is the linear energy
+    coefficient , :math:`\\vec{n}` is the normal vector to the plane
+    (`plane_normal`), and :math:`\\vec{p}` is the plane origin (`plane_origin`):
+
+
+    .. rubric:: Example
+
+    .. code-block:: python
+
+        linear = hoomd.hpmc.external.Linear()
+        linear.alpha['A'] = 0.2
+        simulation.operations.integrator.external_potentials = [linear]
+
+    .. py:attribute:: alpha
+
+        The linear energy coefficient :math:`\\alpha` by particle type.
+
+        Type: `TypeParameter` [`tuple` [``particle_type``, ``particle_type``],
+        `float`]
+
+    .. py:attribute:: plane_origin
+
+        A point on the plane where the energy is zero.
+
+        Type: (`float`, `float`, `float`)
+
+    .. py:attribute:: plane_normal
+
+        A unit length vector perpendicular to the plane where the energy is
+        zero.
+
+        Type: (`float`, `float`, `float`)
+    """
+    _cpp_class_name = "ExternalPotentialLinear"
+
+    def __init__(self,
+                 default_alpha=None,
+                 plane_origin=(0, 0, 0),
+                 plane_normal=(0, 1, 0)):
```

Comment:
```suggestion
                 plane_normal=(0, 0, 1)):
```
I imagine the canonical use case of this potential will be to simulate gravity, which most people would intuitively think about being in the z direction.

Reply:
Right. When a user interested in using this for gravity is running a 2D simulation, it needs to be (0, 1, 0).